dsm/resource_sobject.go

```diff
@@ -183,15 +187,36 @@ func unmarshalStringToJson(inputString string) (interface{}, error) {
 func createSO(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 	endpoint := "crypto/v1/keys"
+	key_size := d.Get("key_size").(int)
+	obj_type := d.Get("obj_type").(string)
+	elliptic_curve := d.Get("elliptic_curve").(string)
 
 	security_object := map[string]interface{}{
 		"name":        d.Get("name").(string),
-		"obj_type":    d.Get("obj_type").(string),
-		"key_size":    d.Get("key_size").(int),
+		"obj_type":    obj_type,
 		"group_id":    d.Get("group_id").(string),
 		"description": d.Get("description").(string),
 	}
 
+	if obj_type == "EC" {
+		if key_size > 0 || len(elliptic_curve) == 0 {
```

Comment:
what will happen if user specifies neither key_size nor elliptic_curve as both are optional fields? Will it be error?

Reply:
In case of EC it will show: key_size should not be specified and elliptic_curve should be specified for EC. 
In case of other keys it will show: key_size should be specified and elliptic_curve should not be specified for <specific key>.

Since it is OR condition either if we give one wrong parameter or both it will trigger the error.